pkg/server/catalog/catalog.go

```diff
@@ -215,18 +211,24 @@ func (c *ServerCatalog) categorize() error {
 				return fmt.Errorf("Plugin %s does not adhere to UpstreamCA interface", p.Config.PluginName)
 			}
 			c.upstreamCAPlugins = append(c.upstreamCAPlugins, NewManagedUpstreamCA(pl, p.Config))
+		case KeyManagerType:
+			pl, ok := p.Plugin.(keymanager.KeyManager)
+			if !ok {
+				return fmt.Errorf("Plugin %s does not adhere to KeyManager interface", p.Config.PluginName)
+			}
+			c.keyManagerPlugins = append(c.keyManagerPlugins, NewManagedKeyManager(pl, p.Config))
+
 		default:
 			return fmt.Errorf("Unsupported plugin type %s", p.Config.PluginType)
 		}
 	}
 
 	// Guarantee we have at least one of each type
```

Comment:
Would be good to update this comment
`// Guarantee we have at least one of each type we require` or something